code/core/src/router/utils.ts

```diff
@@ -155,13 +163,33 @@ const encodeSpecialValues = (value: unknown): any => {
   return value;
 };
 
-const QS_OPTIONS: IStringifyOptions = {
-  encode: false, // we handle URL encoding ourselves
+const QS_OPTIONS: Partial<QueryOptions> = {
   delimiter: ';', // we don't actually create multiple query params
-  allowDots: true, // encode objects using dot notation: obj.key=val
-  format: 'RFC1738', // encode spaces using the + sign
-  serializeDate: (date: Date) => `!date(${date.toISOString()})`,
+  nesting: true,
+  nestingSyntax: 'js', // encode objects using dot notation: obj.key=val
 };
+
+// Replaces some url-encoded characters with their decoded equivalents.
+// The URI RFC specifies these should be encoded, but all browsers will
+// tolerate them being decoded, so we opt to go with it for cleaner looking
+// URIs.
+const decodeKnownQueryChar = (chr: string) => {
+  switch (chr) {
+    case '%20':
+      return '+';
```

Comment:
Either way, I don't think it matters right now, because when trying out a Date-control on `next`, a Date arg always turns into the UNIX time number , and never the special value `!date(ISO...)`.

It's probably because of these lines, which have been here since the very beginning: https://github.com/storybookjs/storybook/blob/next/code/lib/blocks/src/controls/Date.tsx/#L93-L96

I'm assuming this is a bug still @ghengeveld?

Reply:
yeah I suspect if we decode `+`, we lose the knowledge of if it was a space or a plus. but maybe that loss is ok since we treat both as a space? 🤔 

I'm not really sure of the best thing to do here. you will know better than I do, so I'm happy to go with whatever you and the others decide